### PR TITLE
fix: Prevent crash due to concurrent map access in Go plugin log processing

### PR DESCRIPTION
**Pull request by quzard**
# Issue Description
A fatal error, "concurrent map read and map write," occurs when using Go plugins under high log volume and concurrent conditions. The root cause of this issue is traced back to the ReceiveLogGroup function in plugin_runner_v1.go. The context passed to subsequent log processing did not undergo a deep copy, leading to the use of the same map object.

# Problematic Scenario
In the aforementioned function, while the context map is being assigned new values, the Add function in aggregator_context.go is concurrently accessing values from the same context. This simultaneous operation on the same map object results in concurrent read and write errors, which is not permissible in Go without proper synchronization mechanisms.

# Resolution
To address this problem, a deep copy of the context map is now implemented for each log processing instance. By doing so, each context uses a separate map object, eliminating the reuse of the same map and thereby preventing concurrent read and write conflicts. This deep-copy approach ensures that each log processor operates on an independent copy of the context, maintaining the integrity of the map access throughout the logging system.

### CODE REVIEW

**Review by messixukejia — CHANGES_REQUESTED** [review]

File: pluginmanager/plugin_runner_v1.go [hunk 1]
```diff
@@ -432,10 +432,11 @@ func (p *pluginv1Runner) ReceiveRawLog(log *pipeline.LogWithContext) {
 }
 
 func (p *pluginv1Runner) ReceiveLogGroup(logGroup pipeline.LogGroupWithContext) {
-	context := logGroup.Context
 	topic := logGroup.LogGroup.GetTopic()
-	context[ctxKeyTopic] = topic
 	for _, log := range logGroup.LogGroup.Logs {
+		context := map[string]interface{}{}
```
[the remark this answers is not in the document]
Comment (quzard) [reply pluginmanager/plugin_runner_v1.go:437]: map在go语言中，是引用类型，当复制的时候，不能用map1 = map2 这样的值类型复制方式。
需要深度copy才行。

[the remark this answers is not in the document]
Comment (linrunqi08) [reply pluginmanager/plugin_runner_v1.go:437]: !p.LogstoreConfig.GlobalConfig.UsingOldContentTag 只有在进入这个分支的时候，才需要深度copy，是不是可以把判断提前一些

File: pluginmanager/plugin_runner_v1.go [hunk 2]
```diff
@@ -432,18 +432,23 @@ func (p *pluginv1Runner) ReceiveRawLog(log *pipeline.LogWithContext) {
...
+			context := map[string]interface{}{}
+			for key, value := range logGroup.Context {
+				context[key] = value
+			}
+			context[ctxKeyTopic] = topic
 			context[ctxKeyTags] = logGroup.LogGroup.LogTags
 			p.ReceiveRawLog(&pipeline.LogWithContext{Log: log, Context: context})
 		} else {
+			context := logGroup.Context
+			context[ctxKeyTopic] = topic
```
Comment (linrunqi08) [inline pluginmanager/plugin_runner_v1.go:451]: 你这里不是还是在写logGroup.Context吗，问题还是会存在

Comment (quzard) [reply pluginmanager/plugin_runner_v1.go:451]: 不会吧，这个logGroup.Context只有这个地方会用写。

**Review by yyuuttaaoo — APPROVED** [review]

**Review by messixukejia — APPROVED** [review]

**Pull request merged**